### PR TITLE
SynthonSpace search timeout

### PR DESCRIPTION
**Pull request by DavidACosgrove**
#### Reference Issue
No issue

#### What does this implement/fix? Explain your changes.
Adds a timeout to the searching.  

#### Any other comments?
This is probably only relevant for a fingerprint search where many hits can be returned, and generating them takes quite a lot of time.

### CODE REVIEW

File: Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h [hunk 1]
```diff
@@ -78,6 +78,8 @@ struct RDKIT_SYNTHONSPACESEARCH_EXPORT SynthonSpaceSearchParams {
              // times, a lower number will give faster searches at the
              // risk of missing some hits.  The value you use should have
              // a positive correlation with your FOMO.
+  long long int timeOut{
```
Comment (bp-kelley) [inline Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h:81]: Any reason not to use

```
std::int64_t
```

Comment (DavidACosgrove) [reply Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h:81]: I guess I hovered over the relevant value in CLion and it told me it was a long long int.

Comment (bp-kelley) [reply Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h:81]: I think std::chrono::duration::count returns mostly "long long int".  I prefer int64_t but YMMV.

Comment (greglandrum) [reply Code/GraphMol/SynthonSpaceSearch/SynthonSpace.h:81]: > Any reason not to use
> 
> ```
> std::int64_t
> ```

I would prefer this too. I like the types that are specific about their size.

**Review by greglandrum — COMMENTED** [review]
I haven't done a full review yet, but a couple of comments:
- I don't feel really strongly about this, but it seems like adding the total search time to the results adds some complexity for very little value (this is an easy thing for people to track themselves if they want to)
- Since the current implementation will only timeout when a new result is found, it's entirely possible that a search returning no (or very few) results will run long past the timeout provided. I think the timeout check cannot be tied to `results.size()`

**Comment by greglandrum** [comment]
> > I haven't done a full review yet, but a couple of comments:
> > 
> > * I don't feel really strongly about this, but it seems like adding the total search time to the results adds some complexity for very little value (this is an easy thing for people to track themselves if they want to)
> 
> I'm not fussed either way. It just seemed an easy thing to capture, given the information available. I've removed it.

From the failing CI builds, it looks like you missed some of the instances.

**Comment by DavidACosgrove** [comment]
Yeah, missed the python wrapper.  Sorry.

**Comment by DavidACosgrove** [comment]
The originally CI build failed due to a timeout, and then life moved on and created a merge conflict.

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
will merge after the CI builds pass

**Pull request merged**